polkadot/runtime/parachains/src/coretime/mod.rs

```diff
@@ -244,11 +288,50 @@ impl<T: Config> Pallet<T> {
 				Location::new(0, [Junction::Parachain(T::BrokerId::get())]),
 				message,
 			) {
-				log::error!("Sending `NotifyCoreCount` to coretime chain failed: {:?}", err);
+				log::error!(target: LOG_TARGET, "Sending `NotifyCoreCount` to coretime chain failed: {:?}", err);
 			}
 		}
 	}
 
+	/// Provide the amount of revenue accumulated from Instantaneous Coretime Sales from Relay-chain
+	/// block number last_until to until, not including until itself. last_until is defined as being
+	/// the until argument of the last notify_revenue message sent, or zero for the first call. If
+	/// revenue is None, this indicates that the information is no longer available. This explicitly
+	/// disregards the possibility of multiple parachains requesting and being notified of revenue
+	/// information.
+	///
+	/// The Relay-chain must be configured to ensure that only a single revenue information
+	/// destination exists.
+	pub fn notify_revenue(when: BlockNumber) -> DispatchResult {
+		let now = <frame_system::Pallet<T>>::block_number();
+		let when_bnf: BlockNumberFor<T> = when.into();
+
+		// When cannot be in the future.
+		ensure!(when_bnf <= now, Error::<T>::RequestedFutureRevenue);
+
+		let claim = <assigner_on_demand::Pallet<T>>::start_revenue_claim_until(when_bnf);
+		log::debug!(target: LOG_TARGET, "Revenue info requested: {:?}", claim.amount);
+
+		let raw_revenue: Balance = claim.amount.try_into().map_err(|_| {
+			log::error!(target: LOG_TARGET, "Converting on demand revenue for `NotifyRevenue` failed");
+			Error::<T>::AssetTransferFailed
+		})?;
+
+		with_transaction(|| -> TransactionOutcome<Result<_, DispatchError>> {
```

Comment:
Why do we call this in a transaction?

Reply:
Because funds are first withdrawn and then checked out, if the withdrawal is successful and the checkout is not, we end up with unbalanced funds.

Reply:
Same as below. If it fails, there is a bug in your XCM code that is there inherently.

Reply:
Okay, I'll remove it.

CC @joepetrowski I remember you insisted this part must be transactional